pkg/config/load.go

```diff
@@ -112,3 +113,36 @@ func OperateOnCIOperatorConfigDir(configDir string, callback func(*cioperatorapi
 		return nil
 	})
 }
+
+func LoggerForInfo(repoInfo FilePathElements) *logrus.Entry {
+	return logrus.WithFields(logrus.Fields{
+		"org":         repoInfo.Org,
+		"repo":        repoInfo.Repo,
+		"branch":      repoInfo.Branch,
+		"source-file": repoInfo.Filename,
+	})
+}
+
+type Info struct {
+	Configuration cioperatorapi.ReleaseBuildConfiguration
+	RepoInfo      FilePathElements
+}
+
+func (i *Info) Logger() *logrus.Entry {
+	return LoggerForInfo(i.RepoInfo)
+}
+
+func (i *Info) CommitTo(dir string) {
+	raw, err := yaml.Marshal(i.Configuration)
+	if err != nil {
+		i.Logger().WithError(err).Error("failed to marshal output CI Operator configuration")
+		return
+	}
+	outputFile := path.Join(
+		dir, i.RepoInfo.Org, i.RepoInfo.Repo,
+		fmt.Sprintf("%s-%s-%s.yaml", i.RepoInfo.Org, i.RepoInfo.Repo, i.RepoInfo.Branch),
```

Comment:
1. This ignores variants
2. `FilePathElements` has filename, why wasn't that used? This is probably also an opportunity to make that member a method instead of an attribute.